spec/unit/client-request-cache.spec.js

```diff
@@ -0,0 +1,120 @@
+            return crc.get("athing").then(() => {
+                return crc.get("athing");
+            }).then((res) => {
+                expect(requestCount).toBe(1);
+                expect(res).toBe("Behold, the *thing*");
+            });
+        });
+        it("should expire old items", () => {
+            let requestCount = 0;
+            const crc = new ClientRequestCache(50, 2, () => {
+                requestCount++;
+                return "Behold, the *thing*";
+            });
+            return crc.get("athing").then(() => {
+                return Promise.delay(100);
+            }).then(() => {
+                return crc.get("athing");
+            }).then((res) => {
+                expect(requestCount).toBe(2);
+                expect(res).toBe("Behold, the *thing*");
+            });
+        });
+        it("should hold multiple items", () => {
+            const crc = new ClientRequestCache(1000, 2, (thing) => {
+                if (thing === "1") {
+                    return "Thing 1!";
+                }
+                return "Thing 2!";
+            });
+            
+            return crc.get("1").then((res) => {
+                expect(res).toBe("Thing 1!");
+                return crc.get("2");
+            }).then((res) => {
+                expect(res).toBe("Thing 2!");
+            });
+        });
+        it("should pass down failures", () => {
+            const crc = new ClientRequestCache(1000, 1, () => {
+                return Promise.reject("Sorry, this test has subject to a GDPR request.");
+            });
+            return crc.get("1").then((res) => {
+                fail("Didn't reject");
+            }).catch((err) => {
+                expect(err).toBe("Sorry, this test has subject to a GDPR request.");
+            });
+        });
+        it("should pass args", () => {
+            const crc = new ClientRequestCache(1000, 1, (key, ...args) => {
+                return args;
+            });
+            return crc.get("1", "Hey", "that's", "pretty", "cool").then((res) => {
+                expect(res).toEqual(["Hey", "that's", "pretty", "cool"]);
+            });
+        });
+        it("should reject non-string keys", () => {
+            const crc = new ClientRequestCache(1000, 1, () => { });
+            expect(() => { crc.get({}) }).toThrow();
+            expect(() => { crc.get(1) }).toThrow();
+            expect(() => { crc.get(true) }).toThrow();
+            expect(() => { crc.get(null) }).toThrow();
+            expect(() => { crc.get() }).toThrow();
+        });
+        it("should respect max size", () => {
+            const crc = new ClientRequestCache(1000, 2, (key) => `${key}baa`);
+            return Promise.all([
+                crc.get("1"),
+                crc.get("2"),
+                crc.get("3"),
+                crc.get("4"),
+            ]).then(() => {
+                expect([...crc._requestContent.values()].map((v) => v.content)).toEqual([
```

Comment:
A public `getAllItems` or similar function would be preferred instead of a poke at internal properties.